### PR TITLE
Added L0_stability_perf

### PR DESCRIPTION
**Pull request by aramesh7**
* Added perf stability test to compare model analyzer against perf analyzer standalone
* Fixed bug where early exit from model analyzer might leave tritonserver running.

### CODE REVIEW

File: qa/L0_stability_perf/check_results.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    def check_perf_stability(self):
+        """
+        Get the perf results from each model analyzer run
+        as well as perf_analyzer run and compare them
+        """
+
+        log_paths = glob.glob(os.path.join(os.getcwd(), f"*.log"))
+
+        # Split the logs up into log pairs model_name: [model_name.test.log, model_name.perf.log]
```
Comment (Tabrizian) [inline qa/L0_stability_perf/check_results.py:46]: Could you please explain what's the difference between `model_name.test.log` and `model_name.perf.log`?

File: qa/L0_stability_perf/check_results.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                    latencies.append(float(p99_latency.group(1)))
+                else:
+                    print(f"\n***\n*** No throughput found in {log} \n***")
+                    return False
+
+            # Once all throughputs and latencies are collected, compute diff percentage
+            throughput_diff = 100 * (throughputs[0] -
+                                     throughputs[1]) / throughputs[0]
+
+            latency_diff = 100 * (latencies[1] - latencies[0]) / latencies[1]
```
Comment (Tabrizian) [inline qa/L0_stability_perf/check_results.py:81]: Why throughput diff is calculated different than latency diff?

File: qa/L0_stability_perf/check_results.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+            if throughput_diff > self._tolerance_percent and latency_diff > self._tolerance_percent:
+                print(
+                    f"\n***\n*** Model Analyzer throughput and latency differ "
+                    f"greatly from perf analyzer standalone for model {model_name}. "
+                    f"\n*** perf analyzer values: Throughput={throughputs[0]} infer/sec, p99 Latency={latencies[0]} usec. \n***"
```
Comment (Tabrizian) [inline qa/L0_stability_perf/check_results.py:87]: Maybe include the MA results here as well?

**Review by Tabrizian — APPROVED** [review]

File: qa/L0_stability_perf/check_results.py (hunk 2 above)
Comment (aramesh7) [reply qa/L0_stability_perf/check_results.py:81]: The logic here is that the model-analyzer value can be higher (better than) the perf_analyzer value, but should not be very much worse.

File: qa/L0_stability_perf/check_results.py (hunk 3 above)
Comment (aramesh7) [reply qa/L0_stability_perf/check_results.py:87]: The MA results are appended on error

File: qa/L0_stability_perf/check_results.py (hunk 1 above)
Comment (aramesh7) [reply qa/L0_stability_perf/check_results.py:46]: .test.log is the model-analyzer log, same as all the other tests. .perf.log is the perf analyzer log.

File: qa/L0_stability_perf/check_results.py (hunk 2 above)
Comment (aramesh7) [reply qa/L0_stability_perf/check_results.py:81]: Better values for latency are lower, which is why the sign is flipped.

File: qa/L0_stability_perf/check_results.py (hunk 1 above)
Comment (Tabrizian) [reply qa/L0_stability_perf/check_results.py:46]: Would be great to add a comment but I was just wondering which files they refer to.

**Pull request merged**